### ISSUE TITLE
Avoid Python minor version in hook scripts shebangs

### ISSUE THREAD

**Issue by cansjt** [pre-commit/pre-commit#2260]
### describe your issue

When you run pre-commit install the script(s) installed in `.git/hooks` often include a the minor version of python. E.g. the script starts with:
```
#/usr/bin/env python3.10
```
This is damaging in some use cases. For example, it prevents using git worktree. Let's assume a new release of Python has been release and I want to check whether my code is portable or not, amend it if not, but I still need to work on some features in parallel (upgrading to the next Python version is of lower priority to me):
```
cd /path/to/my-project-trunk
git worktree add ../my-project-3-11 master
cd ../my-project-3-11/
pyenv local 3.11.0
mkvirtualenv -a . --python="$(pyenv which python)" my-project-3-11
pip install -e '.[dev]'
pytest 
# amend my code
git commit -m "Add Python 3.11 support"
```
Problem arises here: `pre-commit` will complain that Python 3.10 is not installed (assuming this was the version used to work on the `my-project-trunk` worktree). Because that version was hard-coded in the scripts shebang.

Removing the `.10` from the script(s) shebangs, solves the issue.

### pre-commit --version

pre-commit 2.17.0

### .pre-commit-config.yaml

```yaml
---
repos:
- repo: https://github.com/pre-commit/pre-commit-hooks
  rev: v4.1.0
  hooks:
  - id: trailing-whitespace
  - id: end-of-file-fixer
  - id: check-yaml
    args: ['--allow-multiple-documents']
    exclude: k8s/.*/templates/.*\.yaml
  - id: debug-statements
- repo: https://github.com/pre-commit/mirrors-isort
  rev: v5.10.1
  hooks:
  - id: isort
- repo: https://gitlab.com/pycqa/flake8
  rev: 3.9.2
  hooks:
  - id: flake8
    files: '.*\.py'
- repo: https://github.com/syntaqx/git-hooks
  rev: v0.0.17
  hooks:
  - id: circleci-config-validate
  - id: shellcheck
    files: '.*\.sh'
- repo: https://github.com/adrienverge/yamllint.git
  rev: v1.26.3
  hooks:
  - id: yamllint
    files: ^.*\.(yaml(lint)?|yml)$
    args: [-c=.yamllint]
- repo: local
  hooks:
  - id: render
    name: Render Helm Templates
    files: ^k8s/.*/.*$
    language: script
    entry: ./render.sh
  - id: branch-up-to-date
    name: Ensure Branch is up-to-date (rebased)
    entry: ./git-branch-uptodate.sh
    language: script
    stages:
    - push
...
```

**Comment by asottile** [comment]
can't reproduce, show me what you're seeing with commands and outputs

**Comment by cansjt** [comment]
It looks like pre-commit no longer installs a python script but a bash script, in `.git/hooks`. Maybe I should have uninstalled and reinstalled `pre-commit`. I did a `pre-commit autoupdate` but I guess that does not update the hook scripts themselves.

**Comment by asottile** [comment]
`autoupdate` has nothing to do with the hook scripts, it's to update versions in the config file.  `pre-commit install` is the command you're looking for
